### PR TITLE
Patched GetInput to stop polling if no input is available

### LINKED ISSUE #121: High CPU utilization on idle
I've noticed some high CPU utilization in idle, when the server is in hibernate. According to `htop` msh seems to be hammering one thread/core constantly. I'm using `v2.4.4`. It seems to increase the following areas: system (probably via syscalls), IRQs and userspace.

### PR DESCRIPTION
**Pull request by cromefire**
Fixes #121

It fixes the issue by stopping the polling for input when an EOF error occurs.

### CODE REVIEW

File: lib/input/input.go [hunk 1]
```diff
@@ -20,6 +21,12 @@ func GetInput() {
 	for {
 		line, err = reader.ReadString('\n')
 		if err != nil {
+			// if stdin is unavailable (msh running as service)
+			// exit from input goroutine to avoid an infinite loop
+			if err == io.EOF {
+				errco.LogMshErr(errco.NewErr(errco.INPUT_UNAVAILABLE_ERROR, errco.LVL_D, "GetInput", "stdin unavailable, exiting input goroutine"))
```
Comment (cromefire) [inline lib/input/input.go:27]: I'd put it on LVL_B as I think it'd be important to know even if not in debug mode (equivalent to info/warn in normal log levels).

**Comment by gekigek99** [comment]
thanks for the pr.
I modified error logging to log the error with errco package and added comments.
can you check that the comment is correct?

**Comment by cromefire** [comment]
It's correct, I'd only change the loglevel as noted in the comment.

File: lib/input/input.go (hunk 1 above)
Comment (gekigek99) [reply lib/input/input.go:27]: I don't think this is the case, on LVL_B there are only logs for minecraft server status, config loading and os support.

Even tho this bugfix is extremely important, the input goroutine exiting should only be reported as LVL_D (deep logging) (LVL_C is reserved for minecraft server logs)

if users dont know why the input functionality disappears, they can change the log level

**Comment by gekigek99** [comment]
I was writing the response to your concern.
In the future I might add the feature of logging to a file and might change a bit the situation

**Comment by gekigek99** [comment]
mmmm yeah you are right... would you mind changing it to LVL_B again?
and add the comment "in case input goroutine returns abnormally while msh is running in terminal, the user must be notified with errco.LVL_B"

**Comment by cromefire** [comment]
done

**Pull request merged**